src/xia2/cli/cluster_analysis.py

```diff
@@ -0,0 +1,240 @@
+from __future__ import annotations
+
+import copy
+import logging
+import os
+import random
+import sys
+
+import numpy as np
+from jinja2 import ChoiceLoader, Environment, PackageLoader
+
+import iotbx.phil
+from dials.array_family import flex
+from dials.util.multi_dataset_handling import (
+    assign_unique_identifiers,
+    parse_multiple_datasets,
+)
+from dials.util.options import ArgumentParser, flatten_experiments, flatten_reflections
+from dials.util.version import dials_version
+
+import xia2.Handlers.Streams
+from xia2.Modules.Analysis import batch_phil_scope
+from xia2.Modules.MultiCrystalAnalysis import MultiCrystalReport
+from xia2.XIA2Version import Version
+
+logger = logging.getLogger("xia2.cluster_analysis")
+
+mca_phil = iotbx.phil.parse(
+    """
+title = 'xia2 report'
+    .type = str
+prefix = 'xia2'
+    .type = str
+
+seed = 42
+  .type = int(value_min=0)
+
+unit_cell_clustering {
+  threshold = 5000
+    .type = float(value_min=0)
+    .help = 'Threshold value for the clustering'
+  log = False
+    .type = bool
+    .help = 'Display the dendrogram with a log scale'
+}
+
+max_cluster_height_difference = 0.5
+  .type = float
+  .short_caption = "Maximum hight difference between clusters"
+max_output_clusters = 10
+  .type = int
+  .short_caption = "Maximum number of important clusters to be output"
+min_cluster_size = 5
+  .type = int
+  .short_caption = "Minimum number of datasets for an important cluster"
+output_correlation_cluster_number = 0
+  .type = int
+  .short_caption = "Option to output a specific correlation cluster when re-running the code"
+output_cos_cluster_number = 0
+  .type = int
+  .short_caption = "Option to output a specific cos cluster when re-running the code"
+
+output {
+  log = xia2.multi_crystal_analysis.log
```

Comment:
Should this be `log = xia2.cluster_analysis.log`?

Reply:
Yes it should be - this has been changed